dotcom-rendering/src/client/sIndicator.ts

```diff
@@ -0,0 +1,255 @@
+			newElem.id = overallRoundelElementId;
+			document.body.appendChild(newElem);
+			return newElem;
+		})();
+
+	if (status === 'hide') {
+		roundelElement.remove();
+		return;
+	}
+
+	roundelElement.style.position = 'fixed';
+	roundelElement.style.bottom = '20px';
+	roundelElement.style.left = '20px';
+	roundelElement.style.zIndex = '79999';
+	roundelElement.style.overflow = 'hidden';
+	roundelElement.style.padding = '10px';
+	roundelElement.style.fontSize = '18px';
+	roundelElement.style.lineHeight = '23px';
+	roundelElement.style.textAlign = 'center';
+	roundelElement.style.borderRadius = '7px';
+	roundelElement.style.width = '40px';
+	roundelElement.style.height = '40px';
+
+	switch (status) {
+		case 'loading':
+			roundelElement.title = 'Checking syndication status...';
+			roundelElement.style.backgroundColor = 'lightblue';
+			// TODO add a spinner
+			break;
+		case 'syndicatable':
+			roundelElement.title = 'This content is syndicatable';
+			roundelElement.style.backgroundColor = 'green';
+			roundelElement.textContent = '✅';
+			break;
+		case 'not-syndicatable':
+			roundelElement.title = 'This content is NOT syndicatable';
+			roundelElement.style.backgroundColor = 'red';
+			roundelElement.textContent = '⛔';
+			break;
+		case 'error':
+			roundelElement.title =
+				'There was an error checking syndication status. Click to check your CAPI key...';
+			roundelElement.style.backgroundColor = 'yellow';
+			roundelElement.textContent = '⚠️';
+			roundelElement.style.cursor = 'pointer';
+			roundelElement.addEventListener(
+				'click',
+				() =>
+					(window.location.href =
+						window.location.hostname === 'localhost'
+							? 'http://localhost:4200/sIndicator'
+							: `https://syndication.${window.location.hostname}/sIndicator`),
+			);
+			break;
+	}
+};
+
+const augmentOnwardAnchor =
+	(syndicatable: boolean) => (element: HTMLElement) => {
+		const shadowColour = syndicatable ? 'green' : 'red';
+		const shadowPosition =
+			element.parentElement &&
+			getComputedStyle(element.parentElement).getPropertyValue(
+				'overflow',
+			) === 'hidden'
+				? 'inset'
+				: '';
+		element.style.boxShadow = `${shadowPosition} 0 0 4px 3px ${shadowColour}`;
+		if (
+			getComputedStyle(element).getPropertyValue('display') === 'inline'
+		) {
+			element.style.display = 'inline-block';
+		}
+	};
+
+export const sIndicator = async (): Promise<void> => {
+	console.log('sIndicator running!');
+
+	if (!maybeSIndicatorCapiKey) {
+		console.error(
+			"sIndicator: no CAPI key found - sIndicator shouldn't be running",
+		);
+		return;
+	}
+
+	displaySIndicatorRoundel('loading');
+
+	const capiKey = maybeSIndicatorCapiKey;
+
+	const pathname = window.location.pathname;
+
+	const maybeProxiedStartingPoint = pathname.indexOf(
+		'https://www.theguardian.com',
+	); // for local dev
+
+	const capiIdFromPath =
+		maybeProxiedStartingPoint === -1
+			? pathname
+			: pathname.substring(maybeProxiedStartingPoint + 28);
```

Comment:
worth a brief comment to explain the 28?

Reply:
good idea, thanks - will do now